chromium_src/components/update_client/update_checker.cc

```diff
@@ -0,0 +1,103 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "../../../../components/update_client/update_checker.cc"
+
+namespace update_client {
+
+SequentialUpdateChecker::SequentialUpdateChecker(
+    scoped_refptr<Configurator> config,
+    PersistedData* metadata)
+    : config_(config), metadata_(metadata) {
+  VLOG(3) << "SequentialUpdateChecker";
+}
+
+SequentialUpdateChecker::~SequentialUpdateChecker() {
+  VLOG(3) << "> ~SequentialUpdateChecker";
+  DCHECK(thread_checker_.CalledOnValidThread());
+  VLOG(3) << "< ~SequentialUpdateChecker";
+}
+
+void SequentialUpdateChecker::CheckForUpdates(
+    const std::string& session_id,
+    const std::vector<std::string>& ids_checked,
+    const IdToComponentPtrMap& components,
+    const base::flat_map<std::string, std::string>& additional_attributes,
+    bool enabled_component_updates,
+    UpdateCheckCallback update_check_callback) {
+  VLOG(3) << "> CheckForUpdates";
+
+  DCHECK(thread_checker_.CalledOnValidThread());
+  DCHECK(!ids_checked.empty());
+
+  for (const auto& app_id : ids_checked) {
+    VLOG(3) << ">  * " << app_id;
+    remaining_ids_.push_back(app_id);
+  }
+
+  session_id_ = session_id;
+  components_ = &components;
+  additional_attributes_ = additional_attributes;
+  enabled_component_updates_ = enabled_component_updates;
+  update_check_callback_ = std::move(update_check_callback);
+
```

Comment:
@mherrmann - To avoid the privacy risk of leaking components that the user has installed, it would help if the request for the next component update is initiated after a random delay. Check this: https://github.com/brave/brave-core/blob/612896e4b30cca7352dfe2aa1247e2995f97b580/vendor/bat-native-ledger/src/bat/ledger/internal/common/time_util.cc#L40

Reply:
@jumde I'm sorry, I don't quite understand your comment. As far as I'm aware, the implementation that is currently in Brave already leaks the components that the user has installed. Also, I don't understand how a random delay could mitigate the issue (the update request would still say which components the user has). If I'm not wrong in saying that Brave already leaks the installed components (out of technical necessity), then this PR doesn't make matters worse and my personal preference would be to merge this PR and address your concerns in a follow-up issue. But of course, I'll be happy to be overruled by you or other people more senior at Brave than me.

Reply:
@mherrmann - Randomized delay reduces linkability of that sequential requests for components for a given user.  If you think adding a random delay will be a bigger change, feel free to file a follow-up issue for this.